specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/ipAllocation.json

```diff
@@ -0,0 +1,395 @@
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of IpAllocation resources.",
+            "schema": {
+              "$ref": "#/definitions/IpAllocationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List IpAllocations in resource group": {
+            "$ref": "./examples/IpAllocationListByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "IpAllocationPropertiesFormat": {
+      "properties": {
+        "subnet": {
+          "$ref": "./virtualNetwork.json#/definitions/Subnet",
+          "description": "The Subnet that using the prefix of this IpAllocation resource."
+        },
+        "virtualNetwork": {
+          "$ref": "./virtualNetwork.json#/definitions/VirtualNetwork",
+          "description": "The VirtualNetwork that using the prefix of this IpAllocation resource."
+        },
+        "type": {
+          "$ref": "#/definitions/IpAllocationType",
+          "description": "The type for the IpAllocation."
+        },
+        "prefix": {
+          "type": "string",
+          "description": "The address prefix for the IpAllocation."
+        },
+        "prefixLength": {
+          "type": "integer",
+          "x-nullable": true,
+          "default": 0,
+          "description": "The address prefix length for the IpAllocation."
+        },
+        "prefixType": {
+          "$ref": "./network.json#/definitions/IPVersion",
+          "default": null,
+          "description": "The address prefix Type for the IpAllocation."
+        },
+        "ipamAllocationId": {
+          "type": "string",
+          "description": "The IPAM allocation ID."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "IpAllocation tags."
+        }
+      },
+      "description": "Properties of the IpAllocation."
+    },
+    "IpAllocation": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/IpAllocationPropertiesFormat",
+          "description": "Properties of the IpAllocation."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
```

Comment:
>Resource [](start = 47, length = 8)

this base definition has "location" and "tags".  Is this a tracked resource?  If so, it needs a PATCH operation

Reply:
This is not a tracked resource, the "tags" actually is not resource tag attribute, it's resource.properties.tags for internal usage.

Reply:
After discussion with PM, this fixed by change the tags to AllocationTags

Reply:
That doesn't fix this comment.  You are inheriting a definition that contains a top-level "location" and "tags" property which is not appropriate for a proxy resource

Reply:
Thanks, I check our code and findout that we are a tracked resource, now I add the patch method and code example with a new commit.